ffi/src/expressions/engine_visitor.rs

```diff
@@ -336,6 +389,101 @@ fn visit_expression_struct(
     call!(visitor, visit_struct_expr, sibling_list_id, child_list_id)
 }
 
+fn visit_expression_transform(
+    visitor: &mut EngineExpressionVisitor,
+    transform: &Transform,
+    sibling_list_id: usize,
+) {
+    // Treat the input path like a column expression
+    let mut path_list_id = 0;
+    if let Some(ref name) = transform.input_path {
+        path_list_id = call!(visitor, make_field_list, 1);
+        visit_expression_column(visitor, name, path_list_id);
+    };
+
```

Comment:
I feel like a doc comment like this would have helped me understand the design a bit better.

```suggestion
// In rust, we represent transforms using two hashmaps: 
//   1) Insertions: optional Input field -> Vec<Expression>
//   2) replacement: input field -> Expression
//
// This visitor instead communicates transforms by combining the two maps into a single list of TransformFields that contains all insertions and replacement for each input field.
// 
// Consider the transform:
//  Transform {
//.      insertions { None -> Int(1), "a" -> String("hello")}, 
//.      replacements { "a" -> Int(2) } 
//.  }
// This becomes a list of FieldTransforms that inserts the expression list after the input field, or replaces the input field with the list of expressions. Note that an insertion + replacement can be treated as replacement with a list.
// field_transforms [
//.          FieldTransform {is_replace = false, None -> ExpressionList[Int(1)]} , 
//.          FieldTransform {is_replace = true,   "a" -> ExpressionList[Int(2), String("hello")]
//  ]
```

Reply:
Point taken about doc comment, but all of this anyway changes in 
* https://github.com/delta-io/delta-kernel-rs/pull/1243

If you like what you see there, maybe we can merge this one without worrying too much about the bits that are about to change?

(or, if we decide it's too closely coupled, we could also fold other PR into this one)